src/StructuredLogViewer.Core/Timeline/Timeline.cs

```diff
@@ -1,31 +1,97 @@
+        const string regexCleanupPhase = @"^Cleanup phase took (?'msTime'[0-9]*.[0-9]*)ms.$";
+        readonly Regex cleanupPhase = new Regex(regexCleanupPhase, RegexOptions.Multiline);
+        const string regexTaskTime = @"^Task '(?'filename'.*)' took (?'msTime'([0-9]*\.[0-9]+|[0-9]+))ms.$";
+        readonly Regex TaskTime = new Regex(regexTaskTime, RegexOptions.Multiline);
+
+        // time(C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.24.28218\bin\Hostx86\x86\c1xx.dll)=0.83512s < 985096605139 - 985104956295 > BB [C:\Users\yuehuang\AppData\Local\Temp\123\main36.cpp]
+        // time(C:\Program Files(x86)\Microsoft Visual Studio\2019\Preview\VC\Tools\MSVC\14.24.28218\bin\Hostx86\x86\c2.dll)=0.01935s < 985104875296 - 985105068765 > BB[C: \Users\yuehuang\AppData\Local\Temp\123\main47.cpp]
+        const string regexBTPlus = @"^time\(.*(c1xx\.dll|c2\.dll)\)=(?'msTime'([0-9]*\.[0-9]+|[0-9]+))s \< (?'startTime'[\d]*) - (?'endTime'[\d]*) \>\s*BB\s*\[(?'filename'[^\]]*)\]$";
+        readonly Regex BTPlus = new Regex(regexBTPlus, RegexOptions.Multiline);
+
+        // Lib: Final Total time = 0.00804s < 5881693617253 - 5881693697673 > PB: 143409152 [D:\test\ConsoleApplication2\x64\Debug\ConsoleApplication2.lib] 
+        // note: there is trailing white space from the tool
+        const string regexLibFinalTime = @"^Lib: Final Total time = (?'msTime'([0-9]*\.[0-9]+|[0-9]+))s \< (?'startTime'[\d]*) - (?'endTime'[\d]*) \>\s*PB:\s*[\d]* \[(?'filename'[^\]]*)\]\s*$";
+        readonly Regex libFinalTime = new Regex(regexLibFinalTime, RegexOptions.Multiline);
+
+        //  Pass 1: Interval #1, time = 0.125s
+        //    Wait PDB close: Total time = 0.000s
+        //    Wait type merge: Total time = 0.000s
+        //  Pass 2: Interval #2, time = 0.016s
+        //  Final: Total time = 0.141s
+        const string regexLinkTotalTime = @"^Final: Total time = (?'msTime'[0-9]*.[0-9]*)s$";
+        readonly Regex linkTotalTime = new Regex(regexLinkTotalTime, RegexOptions.Multiline);
+
+        private bool globalBtplus = false;
+        private bool globalLibTime = false;
+        private bool globalLinkTime = false;
+
+        private TimeSpan oneMilliSecond = TimeSpan.FromMilliseconds(1);
 
         public Dictionary<int, Lane> Lanes { get; set; } = new Dictionary<int, Lane>();
 
-        public Timeline(Build build)
+        public Timeline(Build build, bool includeCpp)
         {
-            this.build = build;
+            Populate(build, includeCpp);
 
-            Populate(build);
+            globalBtplus = false;
+            globalLibTime = false;
+            globalLinkTime = false;
         }
 
-        private void Populate(Build build)
+        private void Populate(Build build, bool includeCpp = false)
         {
             build.VisitAllChildren<TimedNode>(node =>
             {
                 if (node is Build)
                 {
+                    if (includeCpp)

```

Comment:
If possible, I'd like to avoid Cpp-specific logic in Timeline.cs. Let's consider moving it to the Analyzers folder.

Reply:
Moved.